vectortile/src/main/scala/geotrellis/vectortile/protobuf/internal/Command.scala

```diff
@@ -0,0 +1,151 @@
+  *           deltas from the current "cursor".
+  *
+  * }}}
+  * The "cursor" is the location of the current vertex being considered,
+  * and it starts at (0,0) for each Feature. As MoveTo and LineTo commands
+  * are read, the cursor moves according the list of parsed delta pairs.
+  * ClosePath does not move the cursor, but may in future versions of the spec.
+  *
+  * Caveats:
+  *  - Point features, whether single or multi, will always consist of a single MoveTo.
+  *  - Any Polygon in a Polygon feature must have a LineTo with a count of at least 2.
+  *  - ClosePath must always have a parameter count of 1.
+  */
+sealed trait Command
+
+/** `MoveTo` signals a series of moves from the current cursor (default of `(0,0)`).
+  * The parameter pairs that follow don't represent a point to move to,
+  * but instead are deltas from the current cursor.
+  */
+case class MoveTo(deltas: Array[(Int,Int)]) extends Command
+
+/** `LineTo` indicates that a LineString should be continued from the current
+  * cursor.
+  */
+case class LineTo(deltas: Array[(Int,Int)]) extends Command
+
+/** Signals the end of a Polygon. Never has parameters, and doesn't move the cursor. */
+case object ClosePath extends Command
+
+/** Contains convenience functions for handling [[Command]]s. */
+object Command {
+  /** Attempt to parse a list of Command/Parameter Integers. */
+  def commands(cmds: Seq[Int]): ListBuffer[Command] = {
+    @tailrec def work(cmds: Seq[Int], curr: ListBuffer[Command]): ListBuffer[Command] = cmds match {
+      case Nil => curr
+      case ns => parseCmd(ns.head) match {
+        case (1,count) => {
+          val (ps,rest) = ns.tail.splitAt(count * 2)
+          val res = new Array[(Int,Int)](count)
+          var i = 0
+
+          while(i < count) {
+            res.update(i, (unzig(ps(i*2)), unzig(ps(i*2+1))))
+
+            i += 1
+          }
+
+          work(rest, curr += MoveTo(res))
+        }
+        case (2,count) => {
+          val (ps,rest) = ns.tail.splitAt(count * 2)
+          val res = new Array[(Int,Int)](count)
+          var i = 0
+
+          while(i < count) {
+            res.update(i, (unzig(ps(i*2)), unzig(ps(i*2+1))))
+
+            i += 1
+          }
+
+          work(rest, curr += LineTo(res))
+        }
+        case (7,_) => work(ns.tail, curr += ClosePath)
+      }
+    }
+
+    work(cmds, new ListBuffer[Command])
+  }
+
+  /** Convert a list of parsed Commands back into their original Command
+    * and Z-encoded Parameter Integer forms.
+    */
+  def uncommands(cmds: Seq[Command]): Seq[Int] = {
+    cmds.flatMap({
+      case MoveTo(ds) => unparseCmd(1, ds.length) +=: params(ds)
+      case LineTo(ds) => unparseCmd(2, ds.length) +=: params(ds)
+      case ClosePath  => ListBuffer(unparseCmd(7, 1))
+    }).toSeq
+  }
+
+  /** Divide a Command Integer into its Command ID and parameter count. */
+  private def parseCmd(n: Int): (Int,Int) = {
+    val cmd = n & 7
+    val count = n >> 3  // or >> ?
```

Comment:
`>>` and `>>>` are logical and arithmetic shift in Scala, but I can't recall which is which. I think initially I was using `>>>`, which turned out to be wrong, and the comment didn't get cleared.